### PR TITLE
[FIXED JENKINS-41987] Check for null return values from InstanceIdentityProvider methods

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-41987](https://issues.jenkins-ci.org/browse/JENKINS-41987)

Otherwise you get a [cryptic error](https://gist.github.com/jglick/6d802b396c412950c8e0518021c94a2f) when `instance-identity` is misconfigured, for example in tests. This fixes the instantiation error to clearly reflect the actual problem.

@reviewbybees

### CODE REVIEW

File: core/src/main/java/jenkins/slaves/JnlpSlaveAgentProtocol4.java [hunk 1]
```diff
@@ -101,7 +101,13 @@
     public JnlpSlaveAgentProtocol4() throws KeyStoreException, KeyManagementException, IOException {
         // prepare our local identity and certificate
         X509Certificate identityCertificate = InstanceIdentityProvider.RSA.getCertificate();
+        if (identityCertificate == null) {
+            throw new KeyStoreException("no X509Certificate found; perhaps instance-identity is missing or too old");
```
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/slaves/JnlpSlaveAgentProtocol4.java]: Maybe it should explicitly mention the protocol. Just to make the error more diagnosable

File: core/src/main/java/jenkins/slaves/JnlpSlaveAgentProtocol4.java [hunk 2]
```diff
@@ -101,7 +101,13 @@
...
+        }
         RSAPrivateKey privateKey = InstanceIdentityProvider.RSA.getPrivateKey();
+        if (privateKey == null) {
+            throw new KeyStoreException("no RSAPrivateKey found; perhaps instance-identity is missing or too old");
```
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/slaves/JnlpSlaveAgentProtocol4.java]: instance-identity **module**

**Review by oleg-nenashev — COMMENTED** [review]
The change itself looks good to me though maybe there should be more info in the error message.

@jglick Could you please create a JIRA issue? Since JNLP4 is going to be enabled by default in the next LTS, so this fix likely deserves backporting.

File: core/src/main/java/jenkins/slaves/JnlpSlaveAgentProtocol4.java (hunk 1 above)
Comment (jglick) [inline core/src/main/java/jenkins/slaves/JnlpSlaveAgentProtocol4.java]: `JnlpSlaveAgentProtocol4`? The error message will clearly include that already.

**Comment by jglick** [comment]
Backporting seems like overkill to me, since this is merely better diagnostic reporting, but as you like.

**Review by oleg-nenashev — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
🐝

**Comment by jglick** [comment]
see also #7514

**Pull request merged**